File: backend/ruby_foundations/exercises/small_problems_101/easy_9/grocery_list.rb
def buy_fruit(fruit_array)
  fruits = []
  fruit_array.each do |fruit, quantity|
    fruits << [fruit] * quantity
  end
  fruits.flatten
end

puts buy_fruit([["apples", 3], ["orange", 1], ["bananas", 2]]) ==
  ["apples", "apples", "apples", "orange", "bananas","bananas"]

File: frontend/javascript/exercises/list_processing/grocery_list.js
function buyFruit(fruits) {
  result = [];
  fruits.forEach(function (fruit) {
    var quantity = fruit[1];
    for (var i = 0; i < quantity; i++) {
      result.push(fruit[0]);
    }
  });
  return result;
}

console.log(buyFruit([['apples', 3], ['orange', 1], ['bananas', 2]]));